app/ide-desktop/lib/dashboard/src/layouts/Settings.tsx

```diff
@@ -72,32 +72,37 @@ export default function Settings() {
     })()
   }, [sessionType, backend])
 
-  let content: React.JSX.Element
+  let content: JSX.Element | null
   switch (settingsTab) {
     case SettingsTab.account: {
-      content = <AccountSettingsTab />
+      content = backend == null ? null : <AccountSettingsTab backend={backend} />
```

Comment:
If backend is null then show loader. Don't pass null.

Reply:
loader makes no sense here, if backend is null we cannot show the tab at all

Reply:
When the backend could be null?

Reply:
> loader makes no sense here, if backend is null we cannot show the tab at all

Do we exclude the tab from the sidebar then?

Reply:
@MrFlashAccount currently yes (iirc). in the future we will probably want to show it under a paywall instead.

Reply:
How's it related?

Reply:
in the PR the backend is refactored out to [a context object](https://github.com/enso-org/enso/blob/377e23da4a29cebc907ac2abb244b73e82133615/app/ide-desktop/lib/dashboard/src/layouts/Settings/settingsData.tsx#L374)

Reply:
how's what related?

Reply:
And at this point I don't understand why the backend could be null

Reply:
@MrFlashAccount true... i was thinking that the tabs *could* be visible for the purpose of showing paywalls. but of course, if `backend == null` then the user is not logged in (so we don't know the user's plan, so no paywalls)

Reply:
Is it possible now to show settings without logging in? Also, I think this case should be handled by the router(if a user is not logged in, redirect him to the login screen). WDYT?

Reply:
> Is it possible now to show settings without logging in?

no idea. i guess *technically* no because it's no longer possible to avoid logging in?

> Also, I think this case should be handled by the router(if a user is not logged in, redirect him to the login screen). WDYT?

depends on whether we ever want to enable the app for users that are not logged in (e.g. custom builds without the cloud backend). not sure what happens with an *expired* session token when the user is physically offline, but we may need to take that into account too.

so tl;dr: *assuming we will never ever need to show settings without logging in* then it makes sense to `invariant` this. otherwise i don't think it hurts to retain this logic.

it's possible that we may be able to make this tidier by using a `Router` and `Layout`s here? thoughts?

Reply:
![CleanShot 2024-05-29 at 12 40 24@2x](https://github.com/enso-org/enso/assets/61194245/fc58216b-5e6f-42d5-a5e6-ed0471a26862) 
![CleanShot 2024-05-29 at 12 41 06@2x](https://github.com/enso-org/enso/assets/61194245/04c5209e-29bb-47aa-b20b-e84be2203a75)

Looks like this is exactly how it's working right now.

Reply:
> no idea. i guess technically no because it's no longer possible to avoid logging in?

Yeah, seems like it's not possible anymore

Reply:
> depends on whether we ever want to enable the app for users that are not logged in (e.g. custom builds without the cloud backend). not sure what happens with an expired session token when the user is physically offline, but we may need to take that into account too.


Yeah, the behavior of the dashboard, when a user goes offline is still unclear... I think we need to revisit this part one day.

Reply:
right, but `remoteBackend` may still be `null` because the user might be in offline mode (`session.type === 'offline'`). not sure where it stands today (see above re: using the app offline inside e.g. company proxy)

so the `Layout`s here would be to distinguish between settings tab that need the backend and those that don't